syntax/blocks.js

```diff
@@ -333,6 +331,50 @@ disambig("pen.setColor", "pen.setHue", function(children, lang) {
   return (last.isInput && last.isColor) || last.isBlock
 })
 
+disambig("microbit.whenGesture", "gdxfor.whenGesture", function(
+  children,
+  lang
+) {
+  for (var i = 0; i < children.length; i++) {
+    var child = children[i]
+    if (child.shape === "dropdown") {
+      var name = child.value
+      // Yes, "when shaken" gdxfor block exists. But microbit is more common.
+      for (let effect of lang.microbitWhen) {
+        if (minifyHash(effect) === minifyHash(name)) {
+          return true
+        }
+      }
+    }
+  }
+  return false
+})
+
+// There are A LOT of tilt related blocks, so turn all into microbit block
+// Technically it is possible to separate them into group A and B:
+// A uses "front-back", and includes microbit and gdxfor
+// B uses "up-down", and includes boost and wedo
+// But not something that is worth trying, probably
+specialCase("boost.whenTilted", () => blocksById["microbit.whenTilted"])
```

Comment:
Do we definitely need these? I think you can control precedence _without_ using `specialCase` by changing the order of the commands in `commands.js`.

Then again, that would be less explicit and so potentially harder to follow.

Reply:
Yeah, I'd prefer explicit declaration. I don't think use of extension block is common enough to cause performance issues.